enhancements/console/dynamic-plugins.md

```diff
@@ -534,6 +602,26 @@ Both `v1` and `v1alpha1` version are supported. `v1alpha1` plugins will get
 converted by the conversion webhook server into `v1` representation.
 Conversion webhook server is part of the `console-operator` pod.
 
+#### Content Security Policy
+For Content Security Policy (CSP) feature to be considered stable:
+* Update the OpenShift official documentation to include detailed guidelines
+on configuring `ConsolePluginCSP` in the `ConsoleDynamicPlugin` CRD, along with
+recommendations.
+* Add CSP feature to release notes.
+* Extend e2e test suit for console-operator repository:
+    * Standart use case, with a single plugin setting valid sources and
+    validating the final CSP result set in the `console-config.yaml` file.
+    * Edge use cases, such as many plugins defining CSPs with overlapping sources,
+    and validates the final merged CSP result, set in the `console-config.yaml` file.
+* Update one console-demo-plugin to use the new `ConsolePluginCSP` field
+in order to demonstrate the usage.
+* Extend integration test suit in console repository, by checking if the
+HTTP Response header contains proper CSP header.
+
+Note: Currently Console uses `Content-Security-Policy-Report-Only` instead of
+`Content-Security-Policy` header. Due to that the browser will only warn about
+Console CSP violations.
```

Comment:
Should we discuss when we decide to turn on enforcement? It's an important date for plugins because they'll need to adopt these changes before it's enforced.

Reply:
Let's talk about this @jhadvig and come up with a plan.

Reply:
@spadgett so when discussed with @vojtechszocs there should be at least 2-3 releases before we flip the switch.
First we need to:
- finish the [epic with the initial implementation ](https://issues.redhat.com/browse/CONSOLE-4273)
- make sure all the current plugins are complying with the CSP (4.19-4.20)
- add CI check
- flip the switch

Probably there will be more steps that we need to meet, this is just how I see right now.